src/logic/map_objects/immovable.cc

```diff
@@ -193,8 +193,18 @@ ImmovableDescr::ImmovableDescr(const std::string& init_descname,
 			                    program_name.c_str(), name().c_str());
 		}
 		try {
-			programs_[program_name] = new ImmovableProgram(
-			   program_name, programs->get_table(program_name)->array_entries<std::string>(), *this);
+			// TODO(GunChleoc): Compatibility, remove after v1.0
+			if (program_name == "program") {
+				log("WARNING: The main program for the immovable %s should be renamed from 'program' "
+				    "to 'main'\n",
+				    name().c_str());
+				programs_["main"] = new ImmovableProgram(
```

Comment:
If there is a program called "main" and another program called "program" then this compatibility name change must be skipped

Reply:
The chance of that is slim, so I'll just add an additional warning. Keeping it as "program" will be just as wrong.